[Typescript] Trans component's additional props

**Describe the bug**
When Trans has a parent component (either because it's set as a prop, or because the configuration has one by default), it forwards its additional props (the one it does not consume) to that object. However in Typescript, this ends up with an error like:
```
error TS2322: Type '{ children: string; style: CSSProperties; parent: string; }' is not assignable to type 'IntrinsicAttributes & TransProps'.
  Property 'style' does not exist on type 'IntrinsicAttributes & TransProps'.
```

**Occurs in react-i18next version**
11.2.5

**To Reproduce**

Create a simple react-i18next Typescript strictly typed app and use a component:
```
<Trans parent="div" style={{color: 'green'}}>
  Hello world, React is super green
</Trans>
```

This works as I expect it to work at runtime (it creates a div, and style the text in green). However Typescript Compiler chokes on it with:
```
error TS2322: Type '{ children: string; style: CSSProperties; parent: string; }' is not assignable to type 'IntrinsicAttributes & TransProps'.
  Property 'style' does not exist on type 'IntrinsicAttributes & TransProps'.
```

**Expected behaviour**

I'd expect to accept the additional props happily. As a first step I would be fine if Trans was not checking those additional props, it's even better if it manages to check their types properly but I doubt it's doable especially if the parent is set in the config.

**OS (please complete the following information):**

- OS: Linux 1adf3346df5d 5.3.0-19-generic #20-Ubuntu SMP Fri Oct 18 09:04:39 UTC 2019 x86_64 GNU/Linux
- Typescript: Version 3.7.2

--- a/src/index.d.ts
+++ b/src/index.d.ts
@@ -29,7 +29,9 @@ declare module 'i18next' {
   }
 }
 
-export interface TransProps extends Partial<WithT> {
+export interface TransProps<E extends Element = HTMLDivElement>
+  extends React.HTMLProps<E>,
+    Partial<WithT> {
   children?: React.ReactNode;
   components?: React.ReactNode[];
   count?: number;
@@ -37,12 +39,12 @@ export interface TransProps extends Partial<WithT> {
   i18n?: i18n;
   i18nKey?: string;
   ns?: Namespace;
-  parent?: React.ReactNode;
+  parent?: string; // used in React.createElement
   tOptions?: {};
   values?: {};
   t?: TFunction;
 }
-export function Trans(props: TransProps): any;
+export function Trans<E extends Element = HTMLDivElement>(props: TransProps<E>): React.ReactElement;
 
 export function useSSR(initialI18nStore: any, initialLanguage: any): void;
 

--- a/test/typescript/Trans.test.tsx
+++ b/test/typescript/Trans.test.tsx
@@ -39,7 +39,7 @@ function ns() {
 }
 
 function parent() {
-  return <Trans parent={<div />}>Foo</Trans>;
+  return <Trans parent="div">Foo</Trans>;
 }
 
 function tOptions() {
@@ -53,4 +53,16 @@ function values() {
 function t() {
   const { t } = useTranslation();
   return <Trans t={t}>Foo</Trans>;
+}
+
+function extraDivProps() {
+  return (
+    <>
+      <Trans parent="div" style={{ color: 'green' }}>
+        Foo
+      </Trans>
+      {/* div is the default parent */}
+      <Trans style={{ color: 'green' }}>Foo</Trans>
+    </>
+  );
 }

--- a/test/typescript/withTranslation.test.tsx
+++ b/test/typescript/withTranslation.test.tsx
@@ -2,6 +2,10 @@ import * as React from 'react';
 import { withTranslation, WithTranslation } from 'react-i18next';
 import { default as myI18n } from './i18n';
 
+/**
+ * @see https://react.i18next.com/latest/trans-component
+ */
+
 interface MyComponentProps extends WithTranslation {
   bar: 'baz';
 }

